modules/test/playwright/env/common.env.ts

```diff
@@ -0,0 +1,171 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {glob} from 'glob';
+import {copyFile, existsSync, mkdirSync, readFileSync} from 'node:fs'
+
+import {resolve} from 'path'
+
+import {executeBashScript} from './bashUtil.env';
+import {executeGradleTask} from './gradleUtil.env';
+import {getPropertiesFromFiles, getPropertyValue, writePropertiesFile} from './propertiesUtil.env';
+
+export function deployProjectClientExtensions() {
+	const clientExtensionsListFile = getPlaywrightProjectDir() + '/env/client-extensions.list';
+
+	if (existsSync(clientExtensionsListFile)) {
+		const clientExtensionsListFileContent = readFileSync(clientExtensionsListFile).toString();
+
+		for (const clientExtension of clientExtensionsListFileContent.split("\n")) {
+			_deployClientExtension(clientExtension);
+		}
+	}
+}
+
+export function deployProjectDeployDir() {
+	const projectDeployDir = getPlaywrightProjectDir() + '/env/deploy/';
+
+	if (existsSync(projectDeployDir)) {
+		const deployDir = getLiferayHome() + '/deploy';
+
+		if (!existsSync(deployDir)){
+			mkdirSync(deployDir, { recursive: true });
+		}
+
+		const deployFiles = glob.sync(projectDeployDir + '/*');
+
+		for (const deployFile of deployFiles) {
+			const deployFileName = deployFile.replace(/.+\/([^\/]+)/, '\$1');
+
+			copyFile(deployFile, deployDir + '/' + deployFileName, (error) => {
+				if (error) {
+					throw error;
+				}
+			});
+		}
+	}
+}
+
+export function deployProjectOSGiModules() {
+	const osgiModulesListFile = getPlaywrightProjectDir() + '/env/osgi-modules.list';
+
+	if (existsSync(osgiModulesListFile)) {
+		const osgiModulesListFileContent = readFileSync(osgiModulesListFile).toString();
+
+		for (const osgiModule of osgiModulesListFileContent.split("\n")) {
+			_deployOSGiModule(osgiModule);
+		}
+	}
+}
+
+export function getLiferayHome() {
+	return resolve(getPlaywrightProperty('liferay.home'))
+}
+
+export function getLiferayPortalDir() {
+	return resolve(getPlaywrightProperty('liferay.portal.dir'));
+}
+
+export function getPlaywrightBaseDir() {
+	return resolve(getPlaywrightProperty('playwright.base.dir'));
+}
+
+export function getPlaywrightProjectDir() {
+	const playwrightProjectConfigFiles = glob.sync(getPlaywrightBaseDir() + '/tests/**/config.ts');
```

Comment:
This is a nice way to detect test projects. In my PR I simply assumed that top level folders where test projects, but this is way more powerful :+1: